meta-client/src/main/java/org/apache/helix/metaclient/api/MetaClientInterface.java

```diff
@@ -113,7 +142,22 @@ public Stat (EntryMode mode, int version) {
    */
   void create(final String key, final T data, final EntryMode mode);
 
-  // TODO: add TTL create and renew API
+  /**
+   * Create an entry of given EntryMode with given key, data, and expiry time (ttl).
+   * The entry will automatically purge when reached expiry time and has no children.
+   * The entry will not be created if there is an existing entry with the same key.
+   * @param key key to identify the entry
+   * @param data value of the entry
+   * @param ttl Time-to-live value of the node in milliseconds.
+   */
+  void createWithTTL(final String key, final T data, final long ttl);
+
+  /**
+   * Renews the specified TTL node adding its original expiry time
+   * to the current time.
+   * @param key key to identify the entry
+   */
```

Comment:
nit: it will be great to document some edge cases behavior if you want them to be consistent. e.g. the node isn't TTL type, or doesn't exist at all.

Reply:
+1 on this.